### ISSUE TITLE
False positive for prefer-regexp-test with Redux Toolkit

### ISSUE THREAD

**Issue by lostfictions** [sindresorhus/eslint-plugin-unicorn#1677]
Hi there, the `prefer-regexp-test` rule is giving me a false positive with code similar to this:

```ts
import { createAction, Action } from '@reduxjs/toolkit'

const increment = createAction<number>('INCREMENT')

function someFunction(action: Action) {
  if (increment.match(action)) {
   // ^ Prefer `RegExp#test(…)` over `String#match(…)`.
  }
}
```

`increment` in this snippet is not a string -- the [`match()` method](https://redux-toolkit.js.org/api/createAction#actioncreatormatch) is part of the Redux Toolkit api and works as a TypeScript type guard.

It's a little unfortunate, but I don't suppose there's any way to avoid this error besides disabling the rule? It would probably need type information, like some of the [@typescript-eslint/eslint-plugin rules](https://typescript-eslint.io/rules/), [eslint-plugin-sonarjs](https://github.com/SonarSource/eslint-plugin-sonarjs) rules, etc.

**Comment by thorn0** [comment]
An option that loosens this rule so that it works only with regexp and string literals might be good enough here.
```js
'foo'.match(re); // fail
foo.match(/re/); // fail
foo.match(re); // pass
```
